airflow/cli/commands/remote_commands/task_command.py

```diff
@@ -211,7 +212,10 @@ def _get_ti(
                 f"run_id or logical_date of {logical_date_or_run_id!r} not found"
             )
         # TODO: Validate map_index is in range?
-        ti = TaskInstance(task, run_id=dag_run.run_id, map_index=map_index)
+        dag_version = DagVersion.get_latest_version(dag.dag_id, session=session)
+        if TYPE_CHECKING:
+            assert dag_version
+        ti = TaskInstance(task, run_id=dag_run.run_id, map_index=map_index, dag_version_id=dag_version.id)
```

Comment:
Since this pattern is used so often, maybe it’s better for `get_latest_version` to only return DagVersion and raise an exception if one cannot be found.

Reply:
Or, maybe TI init can handle this for folks?

Reply:
@jedcunningham, do you mean, if it's not provided, use the latest dag version ID in init? The only thing is that session is not provided in init.